packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1122,10 +1122,13 @@ class _TabBarViewState extends State<TabBarView> {
       _controller.animation.addListener(_handleTabControllerAnimationTick);
   }
 
+  List<Widget> ensureGlobalKeysForList(List<Widget> items) =>
+      List<Widget>.generate(items.length, (int index) => KeyedSubtree(key: GlobalKey(), child: items[index]));
```

Comment:
Instead of GlobalKey, could this be a ValueKey with the list-index as value?

Reply:
I was playing around, it seems like uniquekey cannot achieve the goal, I haven't tried value key yet, but I don't expect it to work. I can dig more into it to see why only globalkey work. will update this comment once I sort it out